pkg/ottl/grammar.go

```diff
@@ -296,9 +296,9 @@ func (f *field) accept(v grammarVisitor) {
 }
 
 type key struct {
-	String     *string          `parser:"'[' (@String "`
-	Int        *int64           `parser:"| @Int"`
-	Expression *mathExprLiteral `parser:"| @@ ) ']'"`
+	String     *string `parser:"'[' (@String "`
+	Int        *int64  `parser:"| @Int"`
+	Expression *value  `parser:"| @@ ) ']'"`
```

Comment:
Thanks for looking into this! Would it be possible to restrict the types to `*mathExprLiteral` and `*mathExpression`?

Reply:
adapted, please have a look